schedule/starter/main.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/pborman/uuid"
+	"github.com/temporalio/samples-go/schedule"
+	"go.temporal.io/api/enums/v1"
+	"go.temporal.io/sdk/client"
+)
+
+func main() {
+	ctx := context.Background()
+	// The client is a heavyweight object that should be created once per process.
+	c, err := client.Dial(client.Options{
+		HostPort: client.DefaultHostPort,
+	})
+	if err != nil {
+		log.Fatalln("Unable to create client", err)
+	}
+	defer c.Close()
+	// This schedule ID can be user business logic identifier as well.
+	scheduleID := "schedule_" + uuid.New()
+	workflowID := "schedule_workflow_" + uuid.New()
+	// Create the schedule, start with no spec so the schedule will not run.
+	scheduleHandle, err := c.ScheduleClient().Create(ctx, client.ScheduleOptions{
+		ID:   scheduleID,
+		Spec: client.ScheduleSpec{},
+		Action: &client.ScheduleWorkflowAction{
+			ID:        workflowID,
+			Workflow:  schedule.SampleScheduleWorkflow,
+			TaskQueue: "schedule",
+		},
+	})
+	if err != nil {
+		log.Fatalln("Unable to create schedule", err)
+	}
+	// Delete the schedule once the sample is done
+	defer func() {
+		log.Println("Deleting schedule", "ScheduleID", scheduleHandle.GetID())
+		err = scheduleHandle.Delete(ctx)
+		if err != nil {
+			log.Fatalln("Unable to delete schedule", err)
+		}
+	}()
+	// Manually trigger the schedule once
+	log.Println("Manually triggering schedule", "ScheduleID", scheduleHandle.GetID())
+
+	err = scheduleHandle.Trigger(ctx, client.ScheduleTriggerOptions{
+		Overlap: enums.SCHEDULE_OVERLAP_POLICY_ALLOW_ALL,
+	})
+	if err != nil {
+		log.Fatalln("Unable to trigger schedule", err)
+	}
+	// Update the schedule with a spec so it will run periodically,
+	log.Println("Updating schedule", "ScheduleID", scheduleHandle.GetID())
+	err = scheduleHandle.Update(ctx, client.ScheduleUpdateOptions{
+		DoUpdate: func(schedule client.ScheduleUpdateInput) (*client.ScheduleUpdate, error) {
+			schedule.Description.Schedule.Spec = &client.ScheduleSpec{
+				// Run the schedule at 5pm on Friday
+				Calendars: []client.ScheduleCalendarSpec{
+					{
+						Hour: []client.ScheduleRange{
+							{
+								Start: 16,
```

Comment:
@dnr is this 5pm or am I off by an hour? I wasn't sure what the 0 hour mapped to.